hedera-node/src/test/java/com/hedera/services/store/contracts/precompile/RandomGeneratePrecompiledContractTest.java

```diff
@@ -0,0 +1,176 @@
+				new RunningHash(new Hash(TxnUtils.randomUtf8Bytes(48))));
+
+		final var range = r.nextInt(0, Integer.MAX_VALUE);
+		final var result = subject.compute(randomNumberGeneratorInput(range), frame);
+		final var randomNumber = result.toBigInteger().intValue();
+		assertTrue(randomNumber >= 0 && randomNumber < range);
+	}
+
+	@Test
+	void hasExpectedGasRequirement() {
+		given(dynamicProperties.randomGenerateGasCost()).willReturn(GAS_REQUIREMENT);
+		assertEquals(GAS_REQUIREMENT, subject.gasRequirement(argOf(123)));
+	}
+
+	@Test
+	void inputRangeCannotBeNegative() {
+		final var input = randomNumberGeneratorNegativeInput(-10);
+		final var ex = assertThrows(InvalidTransactionException.class, () -> subject.compute(input, frame));
+		assertEquals(INVALID_RANDOM_GENERATE_RANGE, ex.getResponseCode());
+	}
+
+	@Test
+	void selectorMustBeFullyPresent() {
+		final var fragmentSelector = Bytes.of(0xab);
+		assertNull(subject.compute(fragmentSelector, frame));
+	}
+
+	@Test
+	void selectorMustBeRecognized() {
+		final var fragmentSelector = Bytes.of((byte) 0xab, (byte) 0xab, (byte) 0xab, (byte) 0xab);
+		final var input = Bytes.concatenate(fragmentSelector, Bytes32.ZERO);
+		assertNull(subject.compute(input, frame));
+	}
+
+	@Test
+	void invalidHashReturnsSentinelOutputs() {
+		given(runningHashLeaf.getNMinus3RunningHash()).willReturn(
+				new RunningHash(new Hash()));
+		var result = subject.compute(random256BitGeneratorInput(), frame);
+		assertEquals(32, result.toArray().length);
+
+		// hash is null
+		given(runningHashLeaf.getNMinus3RunningHash()).willReturn(
+				new RunningHash(null));
+		result = subject.compute(random256BitGeneratorInput(), frame);
+		assertNull(result);
+
+		final var range = r.nextInt(0, Integer.MAX_VALUE);
+		result = subject.compute(randomNumberGeneratorInput(range), frame);
+		final var randomNumberBytes = result;
+		assertNull(randomNumberBytes);
+	}
+
+	@Test
+	void interruptedExceptionReturnsNull() throws ExecutionException, InterruptedException {
+		final var runningHash = mock(RunningHash.class);
+		final var futureHash = mock(StandardFuture.class);
+		given(runningHashLeaf.getNMinus3RunningHash()).willReturn(runningHash);
+		given(runningHash.getFutureHash()).willReturn(futureHash);
+		given(futureHash.get()).willThrow(InterruptedException.class);
+
+		var result = subject.compute(random256BitGeneratorInput(), frame);
+		assertNull(result);
+
+		final var range = r.nextInt(0, Integer.MAX_VALUE);
+		result = subject.compute(randomNumberGeneratorInput(range), frame);
+		assertNull(result);
+	}
+
+	private static Bytes random256BitGeneratorInput() {
+		return input(RANDOM_256_BIT_GENERATOR_SELECTOR, Bytes.EMPTY);
+	}
+
+	private static Bytes randomNumberGeneratorInput(final int range) {
+		return input(RANDOM_NUM_IN_RANGE_GENERATOR_SELECTOR, Bytes32.leftPad(Bytes.ofUnsignedInt(range)));
+	}
+
+	private static Bytes randomNumberGeneratorNegativeInput(final int range) {
```

Comment:
Don't `randomNumberGeneratorInput()` and  `randomNumberGeneratorNegativeInput()` effectively do the same  things.
Taking in numbers and returning the random bytes.
This method doesn't really ensure that the result is negative, so i wonder if that numbers need to be embedded here and reuse the `randomNumberGeneratorInput()` logic

Reply:
yeah , makes sense. Merged two methods to one.